code/modules/crafting/metalwork/metalwork_items.dm

```diff
@@ -7,52 +15,45 @@
 	atom_flags = ATOM_FLAG_OPEN_CONTAINER
 	w_class = ITEM_SIZE_NO_CONTAINER
 	material_alteration = MAT_FLAG_ALTERATION_COLOR | MAT_FLAG_ALTERATION_NAME
-// TODO: storage datum when storage PR is merged.
-//	max_w_class = ITEM_SIZE_LARGE
-//	max_storage_space = BASE_STORAGE_CAPACITY(ITEM_SIZE_LARGE)
+	storage = /datum/storage/crucible
+
 	var/max_held = 10
 
 /obj/item/chems/crucible/attackby(obj/item/W, mob/user)
 
+	// Fill a mould.
 	if(istype(W, /obj/item/chems/mould))
 		if(W.material?.hardness <= MAT_VALUE_MALLEABLE)
 			to_chat(user, SPAN_WARNING("\The [W] is currently too soft to be used as a mould."))
 			return TRUE
 		if(standard_pour_into(user, W))
 			return TRUE
 
-	if(istype(W, /obj/item/debris) || istype(W, /obj/item/stack/material))
-
-		if(length(contents) >= max_held)
-			to_chat(user, SPAN_WARNING("\The [src] is full."))
-			return TRUE
+	// Skim off any slag.
+	if(istype(W, /obj/item/chems) && ATOM_IS_OPEN_CONTAINER(W) && W.reagents && !istype(W, /obj/item/chems/food)) // TODO: food not being a container
 
-		var/obj/item/transferring
-		if(istype(W, /obj/item/stack))
-			var/obj/item/stack/input = W
-			if(input.get_amount() <= 5 && user.try_unequip(input))
-				transferring = input
-			else
-				transferring = input.split(5)
-		else if(user.try_unequip(W))
-			transferring = W
+		// Pour contents into the crucible.
+		if(W.reagents.total_volume)
+			var/obj/item/chems/pouring = W
+			if(pouring.standard_pour_into(user, src))
+				return TRUE
 
-		if(transferring)
-			transferring.forceMove(src)
-			visible_message(SPAN_NOTICE("\The [user] drops \the [transferring] into \the [src]."))
+		// Attempting to skim off slag.
+		// TODO: check for appropriate vessel material? Check melting point against temperature of crucible?
+		if(reagents?.total_volume && length(reagents.reagent_volumes) > 1)
+			var/removing = min(amount_per_transfer_from_this, W.reagents.maximum_volume - W.reagents.total_volume)
```

Comment:
i feel like it should use `W.amount_per_transfer_from_this` since we know it's `/obj/item/chems`, i was confused why it kept only transferring 5u
```suggestion
			var/removing = min(W.amount_per_transfer_from_this, REAGENTS_FREE_SPACE(W.reagents))
```

Reply:
hhhhhhuh. i guess, but i would just assume it would use the transfer amount of the thing you're holding... either way i guess it *works*